packages/documentation/src/pages/platform/tools/feature-flags.mdx

```diff
@@ -45,9 +43,80 @@ vagovdev: false
-- Making features or applications available in production should done only as needed for testing with users.
\ No newline at end of file
+- Making features or applications available in production should done only as needed for testing with users.
+
+## Feature flags for Metalsmith code
+
+Va.gov creates some pages based on content from the VA's Drupal CMS. The CMS has its own content model, which can sometimes change. When those changes modify the existing structure of the content model, the queries and templates in vets-website that expect a different model may break. And because the CMS and vets-website are separate systems with different deployment processes, we can't push up changes in both systems simultaneously. In order to keep the two systems in sync, we need to be able to turn features on and off in vets-website depending on what environment we're in, and update that feature state whenever a cms deployment happens.
+
+We've created some infrastructure to make this a little easier to do.
+
+
+### Creating a flag and turning it on or off
+
+In src/site/utilities/featureFlags.js, you'll see a list of current flags and their state:
+
+```js
+// Edit this to add new flags
+const featureFlags = {
+  FEATURE1: 'feature1',
+};
+
+// Edit this to turn flags on or off
+const flagsByBuildtype = {
+  // localhost: [],
+  localhost: [featureFlags.FEATURE1],
+  // vagovdev: [flags.FEATURE1],
+  vagovdev: [],
+  // vagovstaging: [flags.FEATURE1],
+  vagovstaging: [],
+  // vagovprod: [flags.FEATURE1],
+  vagovprod: [],
+};
+```
+
+In these two objects, you can create a new feature flag and set which web environments it is turned on in. Whenever you need to change what environment has a particular feature, you would add the flag to the array associate with that build type.
+
+### Using flags in Node code
+
+There are a couple ways to use the flag in Node Javascript code. One way is to use the `applyFeatureFlags` helper in a file that you would like to create a feature flagged variant of:
+
+```js
+const someQuery = 'some query info';
+
+module.exports = someQuery;
+applyFeatureFlags(module);
+```
+
+This helper looks through all of the enabled feature flags and checks to see if there are files with the same name as the current file, but with a feature flag in the file path before the extension.
+
+So, for example, if your file is `healthCarePage.graphql.js` and there's a feature flag called `feature1`, the helper will look for `healthCarePage.graphql.feature1.js`. If `feature1` is turned on, then it will replace the exported module data with the export from the file with the `feature1` suffix.
+
+If you want to make a smaller conditional logic change than creating a whole new file, you can import the featureFlags file and check the flags directly:
+
+```js
+const { featureFlags, enabledFeatureFlags } = require('./src/site/utilities/featureFlags');
+
+...
+
+if (enabledFeatureFlags[featureFlags.FEATURE1]) {
+  // do something when the flag is on
+}
```

Comment:
It might be nice to make `enabledFeatureFlags` a `Set` at some point.

Reply:
I decided not to do that because I wasn't sure if it would play nice with Liquid. But yeah, that's something we could definitely try out.